plugins/log_request_plugin/src/main.rs

```diff
@@ -0,0 +1,125 @@
+mod smartrest;
+
+use c8y_api::http_proxy::{create_http_client, create_mqtt_client, C8YHttpProxy};
+use c8y_smartrest::smartrest_deserializer::SmartRestLogRequest;
+use c8y_smartrest::topic::{get_log_file_request_done_message, get_log_file_request_executing};
+
+use futures::SinkExt;
+
+use smartrest::read_tedge_logs;
+use std::time::Duration;
+
+use tokio::time::sleep;
+
+const AGENT_LOG_DIR: &str = "/var/log/tedge/agent";
+const MQTT_SESSION_NAME: &str = "log plugin mqtt session";
+const HTTP_SESSION_NAME: &str = "log plugin http session";
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // reading payload from command line arguments
+    let payload = std::env::args().nth(1).expect("no payload given");
+
+    // creating required clients
+    let mut mqtt_client = create_mqtt_client(&MQTT_SESSION_NAME).await?;
+    let mut http_client = create_http_client(&HTTP_SESSION_NAME).await?;
+
+    // retrieve smartrest object from payload
+    let smartrest_obj = SmartRestLogRequest::from_smartrest(&payload)?;
+
+    // 1. set log file request to executing
+    let msg = get_log_file_request_executing().await?;
+    let () = mqtt_client.published.send(msg).await?;
+    // 2. read logs
+    let log_content = read_tedge_logs(&smartrest_obj, AGENT_LOG_DIR)?;
+
+    // 3. upload log file
+    let upload_event_url = http_client.upload_log_binary(&log_content).await?;
+
+    // 4. set log file request to done
+    let msg = get_log_file_request_done_message(&upload_event_url).await?;
+    let () = mqtt_client.published.send(msg).await?;
+
+    // force a pause to allow 4. to run
+    let () = do_one_second_pause().await;
+
+    Ok(())
+}
+
+/// does a one second pause.
+///
+/// NOTE: this is a quick-fix to enable step 4. in main to be executed.
+/// if `do_one_second_pause()` is not called, step 4. in main
+/// does not get triggered.
+async fn do_one_second_pause() {
+    sleep(Duration::from_secs(1)).await;
+}
```

Comment:
One have to add a `connection.close()` method that waits till all the messages are actually sent.

Reply:
I added a follow up task: https://github.com/thin-edge/thin-edge.io/issues/846